packages/functions/src/common/CancelAppointmentHelper.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  getAppointmentsByIds,
+  setAppointment,
+} from "../dal/AppointmentDataAccessLayer";
+import { FunctionsApi } from "@zm-blood-components/common";
+import { getDonor } from "../dal/DonorDataAccessLayer";
+import { notifyOnCancelAppointment } from "../notifications/CancelAppointmentNotifier";
+import * as DbAppointmentUtils from "../utils/DbAppointmentUtils";
+import { DbAppointment } from "../function-types";
+
+export interface ValidCancelAppointmentResponse {
+  isValid: boolean;
+  invalidReason?: string;
+  appointment?: DbAppointment;
+}
+
+export async function cancelAppointment(
+  callerId: string,
+  request: FunctionsApi.CancelAppointmentRequest,
+  shouldNotify: boolean
+) {
+  const donorId = callerId;
+
+  const { isValid, invalidReason, appointment } =
+    await validateCancelAppointment(request.appointmentId, donorId);
+
+  if (!isValid || !appointment) {
+    throw new Error(invalidReason);
+  }
```

Comment:
I don't really like this pattern.

What do you think about:

```
const appointment = getAppointment();

validate(appointment); // If the appointment is invalid, this will throw

// ... continue code here
```

Meaning - there's no reason to return a valid response.
If the appointment is not valid, just throw.